docs/technicalGuide.stories.mdx

```diff
@@ -0,0 +1,299 @@
+
+## Rendering Components with ElementRenderer
+
+`ElementRenderer` forms the core of the message rendering process. It dynamically renders components based on message formats, abstracting away complexity and promoting modularity.
+
+### Benefits of Using ElementRenderer:
+
+**Dynamic Rendering**: `ElementRenderer` enables dynamic rendering of content based on message formats, eliminating the need for manual switching or conditional rendering based on message types.
+
+**Modularity and Extensibility**: By separating the rendering logic into individual components and using a component map, `ElementRenderer` promotes modularity and allows for easy extension or customization of supported formats.
+
+**Consistency and Maintainability**: `ElementRenderer` ensures consistency in rendering across different message formats, making it easier to maintain and update rendering logic centrally.
+
+**Scalability**: As your application grows and evolves, `ElementRenderer` scales with it, accommodating new message formats and components seamlessly.
+
+To use `ElementRenderer`, you need to pass the following props:
+
+```
+<ElementRenderer message={message} supportedElements={supportedElements} />
+```
+
+Similar to `MessageSpace`, all of the supported elements need to be imported. Refer to the [Supported Elements](https://rustic-ai.github.io/ui-components/?path=/docs/documentation-technical-guide--docs#supported-elements), [Message Structure](https://rustic-ai.github.io/ui-components/?path=/docs/documentation-technical-guide--docs#message-structure) and [Format-Specific Data](https://rustic-ai.github.io/ui-components/?path=/docs/documentation-technical-guide--docs#format-specific-data) sections for more information about what to pass as props.
+
+## Supported Elements
+
+Both `MessageSpace` and `ElementRender` need you to pass a list of supported elements to the library. This object is used to match the message format with the correct component. The key should be the format of the message and the value should be the component that will be rendered for the message format. Below are the default supported elements. You could simply copy and paste the list to your code and pass it as `supportedElements` props.
+
+```
+{
+  text: Text,
+  streamingText: StreamingText,
+  markdown: MarkedMarkdown,
+  streamingMarkdown: MarkedStreamingMarkdown,
+  image: Image,
+  timeSeries: RechartsTimeSeries,
+  map: OpenLayersMap,
+  youtubeVideo: YoutubeVideo,
+  table: Table,
+  calendar: FCCalendar,
+}
+```
+
+## Message Structure
+
+If you want to use our library to build your conversational UI interface, you will need to understand the message structure that required by the library.
+
+MessageProps is used to support all of the message formats. Currently, `inReplyTo`, `threadId`, `priority`, `taggedParticipants` and `topicId` are not being used, but we are planing to incorperate them later.
```

Comment:
```suggestion
MessageProps is used to support all of the message formats.
```
I don't think we need to mention this- I would think it all depends on how users want their app's functionality to work. Also, don't we technically use threadId to support updates? Maybe we can mention this below when you're talking about `ThreadableMessage`.